### ISSUE TITLE
generation by unit for germany

### ISSUE THREAD

**Issue by aleave** [EnergieID/entsoe-py#36]
I am trying to query generation by unit for Germany with the following:

    start = pd.Timestamp("20190801", tz="Europe/Brussels")  
    end = pd.Timestamp("20190802", tz="Europe/Brussels")  
    zone_code = "DE"  
    degen = client.query_generation_per_plant(country_code="DE", start=start, end=end)  

but the code returns the following error:
`400 Client Error: Bad Request for url: https://transparency.entsoe.eu/api?documentType=A73&processType=A16&in_Domain=10Y1001A1001A83F&securityToken=35e58724-cbfb-4bf6-b12d-ad8ab9c49171&periodStart=201907312200&periodEnd=201908012200`

Looking in the ENTSOE transparency portal I see that generation by unit in Germany seems to be reported only by control area, not by country total and I wonder if this is the reason why the query fails. If this is the case, I don't see in the documentation how I could query by control area.

An example of what I am trying to retrieve is [here](https://transparency.entsoe.eu/generation/r2/actualGenerationPerGenerationUnit/show?name=&defaultValue=false&viewType=TABLE&areaType=CTA&atch=false&dateTime.dateTime=16.08.2019+00:00|CET|DAYTIMERANGE&dateTime.endDateTime=16.08.2019+00:00|CET|DAYTIMERANGE&area.values=CTY|10Y1001A1001A83F!CTA|10YDE-ENBW-----N&masterDataFilterName=&masterDataFilterCode=&productionType.values=B01&productionType.values=B02&productionType.values=B03&productionType.values=B04&productionType.values=B05&productionType.values=B06&productionType.values=B07&productionType.values=B08&productionType.values=B09&productionType.values=B10&productionType.values=B11&productionType.values=B12&productionType.values=B13&productionType.values=B14&productionType.values=B20&productionType.values=B15&productionType.values=B16&productionType.values=B17&productionType.values=B18&productionType.values=B19&dateTime.timezone=CET_CEST&dateTime.timezone_input=CET+(UTC+1)+/+CEST+(UTC+2)&dv-datatable_length=50):

**Comment by JrtPec** [comment]
Can you try `zone_code = "DE-LU"`?

I think this is related to #9, if you type "DE", it defaults to "DE-AT-LU", but that zone doesn't exist anymore since 2018.

**Comment by aleave** [comment]
@JrtPec I tried with this:
    start = pd.Timestamp('20190801', tz='Europe/Brussels')
    end = pd.Timestamp('20190802', tz='Europe/Brussels')
    degen = client.query_generation_per_plant(country_code='DE-LU', start=start, end=end)

but I still get the same error:
`HTTPError: 400 Client Error: Bad Request for url: https://transparency.entsoe.eu/api?documentType=A73&processType=A16&in_Domain=10Y1001A1001A82H&securityToken=35e58724-cbfb-4bf6-b12d-ad8ab9c49171&periodStart=201907312200&periodEnd=201908012200`

**Comment by Tinkaa** [comment]
Hmm it seems you can indeed only query control area, also noted in the ENTSO-E documentation, even though data for DE-LU is shown on the Transparency platform.

However, the mapping BIDDING_ZONES does contain the control areas in Germany, so could you try using these? You have to set lookup_bzones to True.
E.g. this works for me:
```
zone_code = "DE-TENNET"  
degen = client.query_generation_per_plant(country_code=zone_code, start=start,end=end,lookup_bzones=True)  
```

**Comment by aleave** [comment]
@Tinkaa 
this is getting closer to a solution: your suggestion works for all control area except for Amprion. If I run the following:

    degen_tennet = client.query_generation_per_plant(country_code='DE-TENNET', start=start, end=end, lookup_bzones=True)
    degen_50Hz = client.query_generation_per_plant(country_code='DE-50HZ', start=start, end=end, lookup_bzones=True)
    degen_transnet = client.query_generation_per_plant(country_code='DE-TRANSNET', start=start, end=end, lookup_bzones=True)
    degen_amprion = client.query_generation_per_plant(country_code='DE-AMPRION', start=start, end=end, lookup_bzones=True)

The first three queries work and for the last one I get the following error:

    KeyError: 'DE-AMPRION'

I checked the mappings in `mappings.py` and I see Amprion mapped as `DE-AMPRION` so that is what I am using

**Comment by Tinkaa** [comment]
Apparently there is a typo in TIMEZONE_MAPPINGS for DE-AMPRION. I submitted a pull request to change it, but I don't have the rights to merge that.

**Comment by aleave** [comment]
thanks @Tinkaa 
@JrtPec can you merge ?

**Comment by JrtPec** [comment]
I have merged it and will include it in version 0.2.11.
I'm just waiting a second on #35, to see if I can include that as well.

**Comment by aleave** [comment]
thanks @JrtPec and by the way thanks for all your work on this package.
